tools/pkg/engine_build_configs/test/build_config_test.dart

```diff
@@ -0,0 +1,366 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' as convert;
+
+import 'package:engine_build_configs/src/build_config.dart';
+import 'package:litetest/litetest.dart';
+
+const String buildConfigJson = '''
+{
+  "builds": [
+    {
+      "archives": [
+        {
+          "name": "build_name",
+          "base_path": "base/path",
+          "type": "gcs",
+          "include_paths": ["include/path"],
+          "realm": "archive_realm"
+        }
+      ],
+      "drone_dimensions": ["dimension"],
+      "gclient_variables": {
+        "variable": false
+      },
+      "gn": ["--gn-arg"],
+      "name": "build_name",
+      "ninja": {
+        "config": "build_name",
+        "targets": ["ninja_target"]
+      },
+      "tests": [
+        {
+          "language": "python3",
+          "name": "build_name tests",
+          "parameters": ["--test-params"],
+          "script": "test/script.py",
+          "contexts": ["context"]
+        }
+      ],
+      "generators": {
+        "tasks": [
+          {
+            "name": "generator_task",
+            "parameters": ["--gen-param"],
+            "scripts": ["gen/script.py"]
+          }
+        ]
+      }
+    }
+  ],
+  "generators": {
+    "tasks": [
+      {
+        "name": "global generator task",
+        "parameters": ["--global-gen-param"],
+        "script": "global/gen_script.dart",
+        "language": "dart"
+      }
+    ]
+  },
+  "tests": [
+    {
+      "name": "global test",
+      "recipe": "engine_v2/tester_engine",
+      "drone_dimensions": ["dimension"],
+      "gclient_variables": {
+        "variable": false
+      },
+      "dependencies": ["dependency"],
+      "test_dependencies": [
+        {
+          "dependency": "test_dependency",
+          "version": "git_revision:3a77d0b12c697a840ca0c7705208e8622dc94603"
+        }
+      ],
+      "tasks": [
+        {
+          "name": "global test task",
+          "parameters": ["--test-parameter"],
+          "script": "global/test/script.py"
+        }
+      ]
+    }
+  ]
+}
+''';
+
+int main() {
+  test('BuildConfig parser works', () {
+    final BuildConfig buildConfig = BuildConfig.fromJson(
+      path: 'linux_test_config',
```

Comment:
Does the linux case cover all the bases or should other platforms also be included?

Reply:
it looks like `check_build_configs.sh` is essentially an integration test that will verify ALL the real configs, and run for every commit in the linux_unopt shard.

Reply:
Right. This file is just some unit tests, so the string on this line doesn't matter.